packages/core/src/App/Containers/AccountSwitcher/account-switcher.jsx

```diff
@@ -41,26 +41,48 @@ const AccountSwitcher = props => {
     const [is_dmt5_real_visible, setDmt5RealVisible] = React.useState(true);
     const [is_dxtrade_demo_visible, setDxtradeDemoVisible] = React.useState(true);
     const [is_dxtrade_real_visible, setDxtradeRealVisible] = React.useState(true);
-    const [exchanged_rate, setExchangedRate] = React.useState('');
+    const [exchanged_rate_cfd_real, setExchangedRateCfdReal] = React.useState(1);
+    const [exchanged_rate_demo, setExchangedRateDemo] = React.useState(1);
+    const [exchanged_rate_cfd_demo, setExchangedRateCfdDemo] = React.useState(1);
 
     const wrapper_ref = React.useRef();
     const scroll_ref = React.useRef(null);
 
     const platform_name_dxtrade = getPlatformSettings('dxtrade').name;
     const platform_name_mt5 = getPlatformSettings('mt5').name;
 
+    const account_total_balance_currency = props.obj_total_balance.currency;
+    const cfd_real_currency =
+        props.mt5_login_list.find(account => !isDemo(account))?.currency ||
+        props.dxtrade_accounts_list.find(account => !isDemo(account))?.currency;
+
+    const vrtc_loginid = props.account_list.find(account => account.is_virtual).loginid;
+    const vrtc_currency = props.accounts[vrtc_loginid] ? props.accounts[vrtc_loginid].currency : 'USD';
+    const cfd_demo_currency =
+        props.mt5_login_list.find(account => isDemo(account))?.currency ||
+        props.dxtrade_accounts_list.find(account => isDemo(account))?.currency;
+
     React.useEffect(() => {
         if (getMaxAccountsDisplayed()) {
             setDmt5RealVisible(false);
         }
     }, [getMaxAccountsDisplayed]);
 
     React.useEffect(() => {
-        const vrtc_loginid = props.account_list.find(account => account.is_virtual).loginid;
-
-        getExchangeRate(props.accounts[vrtc_loginid].currency, props.obj_total_balance.currency).then(res =>
-            setExchangedRate(res)
-        );
+        const getCurrentExchangeRate = (currency, setExchangeRate) => {
+            getExchangeRate(currency, account_total_balance_currency).then(res => {
+                setExchangeRate(res);
+            });
+        };
+        if (cfd_real_currency !== account_total_balance_currency) {
+            getCurrentExchangeRate(cfd_real_currency, setExchangedRateCfdReal);
+        }
+        if (vrtc_currency !== account_total_balance_currency) {
+            getCurrentExchangeRate(vrtc_currency, setExchangedRateDemo);
+        }
+        if (cfd_demo_currency !== account_total_balance_currency) {
+            getCurrentExchangeRate(cfd_demo_currency, setExchangedRateCfdDemo);
+        }
```

Comment:
```suggestion
        const getCurrentExchangeRate = ({ currency, callback }) => {
    if (currency !== account_total_balance_currency) {
        getExchangeRate(currency, account_total_balance_currency).then(res => {
            callback(res);
        });
    }
};
[
    {
        currency: cfd_demo_currency,
        callback: setExchangedRateCfdDemo,
    },
    {
        currency: cfd_real_currency,
        callback: setExchangedRateCfdReal,
    },
    {
        currency: vrtc_currency,
        callback: setExchangedRateDemo,
    },
].forEach(getCurrentExchangeRate);
```
good job! above is just an alternative option of how it can be written what do you think? 😊